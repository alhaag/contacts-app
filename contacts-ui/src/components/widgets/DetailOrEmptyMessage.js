// third party
import React, { Component } from 'react'


const DetailOrEmptyMessage = (props) => {

    const { data, emptyText='NÃ£o informado' } = props;

    if (data) {
        return <span>{data}</span>
    }

    return <span className="md-text--theme-secondary">{emptyText}</span>
}

export default DetailOrEmptyMessage
