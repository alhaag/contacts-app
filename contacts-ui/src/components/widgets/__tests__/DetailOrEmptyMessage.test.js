import React from 'react'
import renderer from 'react-test-renderer'
import DetailOrEmptyMessage from '../DetailOrEmptyMessage';

test('Presentation DetailOrEmptyMessage component has data text', () => {
  const component = renderer.create(
    <DetailOrEmptyMessage data="testing text" />
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Presentation DetailOrEmptyMessage component empty data text', () => {
  const component = renderer.create(
    <DetailOrEmptyMessage data={undefined} />
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Presentation DetailOrEmptyMessage component empty data text and custom error', () => {
  const component = renderer.create(
    <DetailOrEmptyMessage data={undefined} emptyText='NÃ£o informado' />
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})