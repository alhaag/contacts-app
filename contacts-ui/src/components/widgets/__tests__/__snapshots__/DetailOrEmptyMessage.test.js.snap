// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presentation DetailOrEmptyMessage component empty data text 1`] = `
<span
  className="md-text--theme-secondary"
>
  Não informado
</span>
`;

exports[`Presentation DetailOrEmptyMessage component empty data text and custom error 1`] = `
<span
  className="md-text--theme-secondary"
>
  Não informado
</span>
`;

exports[`Presentation DetailOrEmptyMessage component has data text 1`] = `
<span>
  testing text
</span>
`;
